### ISSUE TITLE
vacation.pl: friendly_from is not used anymore

### ISSUE THREAD

**Issue by IdahoPL** [postfixadmin/postfixadmin#543]
At some point between vacation.pl 4.0 and 4.2 friendly_from stopped working. It's still declared in 4.2 config but this variable is not used anywhere in the code after that.

I was using it to put From: header in format "Autoresponder for user@domain.tld \<user@domain.tld\>" (with some minor patching on 4.0 version).

**Comment by IdahoPL** [comment]
Maybye something like this?

`
--- vacation.pl     2021-09-22 12:26:58.122967000 +0200
+++ vacation-local.pl  2021-09-22 13:04:39.753693000 +0200
@@ -499,7 +499,7 @@
         $email = Email::Simple->create(
             header => [
                 To      => $to,
-                From    => $from,
+                From    => encode_mimewords($friendly_from . " $from <$from>", 'Charset', 'UTF-8'),
                 Subject => encode_mimewords($subject, 'Charset', 'UTF-8'),
                 Precedence => 'junk',
                 'Content-Type' => "text/plain; charset=utf-8",
`

**Comment by DavidGoodwin** [comment]
what about this : 
```
diff --git a/VIRTUAL_VACATION/vacation.pl b/VIRTUAL_VACATION/vacation.pl
index f8424056..304ff466 100644
--- a/VIRTUAL_VACATION/vacation.pl
+++ b/VIRTUAL_VACATION/vacation.pl
@@ -496,10 +496,15 @@ sub send_vacation_email {
 
         $subject = Encode::encode_utf8($subject) if(Encode::is_utf8($subject));
         $body = Encode::encode_utf8($body) if(Encode::is_utf8($body));
+
+        $email_from = $from;
+        if($friendly_from ne '') {
+            $email_from = encode_mimewords($friendly_from, 'Charset', 'UTF-8') . " <$from>";
+        }
         $email = Email::Simple->create(
             header => [
                 To      => $to,
-                From    => $from,
+                From    => $email_from,
                 Subject => encode_mimewords($subject, 'Charset', 'UTF-8'),
                 Precedence => 'junk',
                 'Content-Type' => "text/plain; charset=utf-8",

```

My logic being ...

a. If someone has $friendly_from set to the default ( '' ) - then continue as we are now
b. If someone has $friendly_form set, then use that as the human-friendly From name (so if `$friendly_from = 'MyCorp AutoResponder';` you'd have a From line like : `From: MyCorp AutoResponder <original@recipient.domain>'`

with your change you'd get : ``From: MyCorp AutoResponder original@recipient.domain <original@recipient.domain>'`

Do people want the repetition of original@recipient.domain in the From line?

**Comment by IdahoPL** [comment]
> Do people want the repetition of [original@recipient.domain](mailto:original@recipient.domain) in the From line?

Yes, Outlook often hides e-mail and show user only "Name Surname".
When From: header is "Name Surname" <original@recipient.domain> then Outlook tends to hide e-mail address and show user only "Name Surname". It's PITA with autoresponder, because Outlook show only "MyCorp AutoResponder" or, in my case, "Autoresponder". I've got numerous requests to add e-mail to it, so I did.

Your patch proposition is better, thank you :) Is it possibe to use some kind of macro to $friendly_from to add (or not if it's not needed) repetition of original@recipient.domain?
